lib/api/listParts.js

```diff
@@ -138,7 +140,22 @@ function listParts(authInfo, request, log, callback) {
         },
         function waterfall3(destBucket, mpuBucket, mpuOverviewObj, next) {
             if (config.backends.data === 'multiple') {
-                const location = mpuOverviewObj.controllingLocationConstraint;
+                let location;
+                // if controlling location constraint is not stored in object
+                // metadata, mpu was initiated in legacy S3C, so need to
+                // determine correct location constraint
+                if (!mpuOverviewObj.controllingLocationConstraint) {
+                    const backendInfoObj = locationConstraintCheck(request,
+                        null, destBucket, log);
+                    if (backendInfoObj.err) {
+                        return process.nextTick(() => {
+                            callback(backendInfoObj.err);
```

Comment:
I think you have to call `next(backendInfoObj.err)` to end the async waterfall properly. Also in other parts of the waterfall I see `return next(err, destBucket)`, there seem to be a use for the extra argument in case of error, so likely better to follow the pattern.

Reply:
👍 good catch